packages/server/src/services/chatflows/index.ts

```diff
@@ -41,35 +41,14 @@ const checkIfChatflowIsValidForStreaming = async (chatflowId: string): Promise<a
         const edges = parsedFlowData.edges
         const { graph, nodeDependencies } = constructGraphs(nodes, edges)
 
-        const endingNodeIds = getEndingNodes(nodeDependencies, graph)
-        if (!endingNodeIds.length) {
-            throw new InternalFlowiseError(StatusCodes.NOT_FOUND, `Ending nodes not found`)
-        }
-
-        const endingNodes = nodes.filter((nd) => endingNodeIds.includes(nd.id))
+        const endingNodes = getEndingNodes(nodeDependencies, graph, nodes)
 
         let isStreaming = false
-        let isEndingNodeExists = endingNodes.find((node) => node.data?.outputs?.output === 'EndingNode')
+        let isEndingNodeExists = endingNodes.some((node) => node.data?.outputs?.output === 'EndingNode')
 
         for (const endingNode of endingNodes) {
```

Comment:
Can we change the naming from `isEndingNodeExists` to `isCustomFunctionEndingNode`? this can better reflects the purpose of this check.

I think once `isEndingNodeExists`, we dont have to run the `for` loop below right? we can directly return `{ isStreaming : false }`